src/languages/typescript.ts

```diff
@@ -200,7 +200,10 @@ const nodeMatchers: Partial<
     )
   ),
   statement: STATEMENT_TYPES.map((type) => `export_statement?.${type}`),
-  condition: conditionMatcher("*[condition]"),
+  condition: cascadingMatcher(
+    patternMatcher("ternary_expression[condition]"),
+    conditionMatcher("*[condition]")
```

Comment:
I'm a bit nervous about relying on order here.  I'm also nervous about using `*` for something that blindly removes parens.  Maybe better to use the exact scope type(s) here?

Reply:
Isn't order the point of the cascading matcher?

We can do that

Reply:
Yeah I guess that's a fair point.  But the I think the blindly removing parens thing holds, esp given the error we saw that led to this PR

Reply:
Have now added specific types